src/models/Zone.java

```diff
@@ -182,15 +182,15 @@ private static boolean containsClock(Clock clock, List<Clock> clocks) {
         return clocks.contains(clock);
     }
 
-    public Guard buildGuardsFromZone(List<Clock> clocks, List<Clock> relevantClocks) {
-        List<Guard> guards = new ArrayList<>();
-        guards.addAll(buildNormalGuardsFromZone(clocks, relevantClocks));
-        guards.addAll(buildDiagonalConstraintsFromZone(clocks, relevantClocks));
-        return new AndGuard(guards);
+    public Expression buildExpressionFromZone(List<Clock> clocks, List<Clock> relevantClocks) {
+        List<Expression> expressions = new ArrayList<>();
+        expressions.addAll(buildNormalGuardsFromZone(clocks, relevantClocks));
+        expressions.addAll(buildDiagonalConstraintsFromZone(clocks, relevantClocks));
```

Comment:
For both the calls to the "build" functions. I think we should prefix them with "create" and remove "FromZone".